malariagen_data/ag3.py

```diff
@@ -649,43 +651,42 @@ def snp_effects(self, transcript, site_mask):
 
         # get feature direct from geneset
         gs = self.geneset()
-        gs.rename(
-            columns={"ID": "feature_id", "Parent": "parent_id", "end": "stop"},
-            inplace=True,
-        )
-        feature = gs[gs.feature_id == transcript].squeeze()
+        feature = gs[gs["ID"] == transcript].squeeze()
 
         # grab pos, ref and alt for chrom arm from snp_sites
-        sites = self.snp_sites(contig=feature.seqid, site_mask=site_mask)
+        pos, ref, alt = self.snp_sites(
+            contig=feature.seqid, site_mask=site_mask, site_filters=site_filters
+        )
+
         # sites are dask arrays, turn pos into sorted index
-        pos = allel.SortedIndex(sites[0].compute())
+        pos = allel.SortedIndex(pos.compute())
+
         # locate transcript range
-        loc = pos.locate_range(feature.start, feature.stop)
+        loc = pos.locate_range(feature.start, feature.end)
+
         # dask compute on the sliced arrays to speed things up
-        ref = sites[1][loc].compute()
-        alt = sites[2][loc].compute()
+        pos = pos[loc]
+        ref = ref[loc].compute()
+        alt = alt[loc].compute()
 
         # build an initial dataframe with contig, pos, ref, alt columns
-        df_in = pandas.DataFrame()
-        df_in["position"] = np.asarray(pos[loc])
-        df_in["ref_allele"] = [q.tobytes().decode() for q in np.asarray(ref)]
-        # bytes within lists within lists...
-        df_in["alt_allele"] = [list(q.tobytes().decode()) for q in list(alt)]
-        # explode the alt alleles into their own rows
-        variants = df_in.explode("alt_allele").reset_index(drop=True)
+        cols = {
+            "position": np.repeat(pos, 3),
+            "ref_allele": np.repeat(ref.astype("U1"), 3),
+            "alt_allele": alt.astype("U1").flatten(),
+        }
+        df_variants = pandas.DataFrame(cols)
+
```

Comment:
Changed to construct initial dataframe the same way as in snp_allele_frequencies().